File: wow-minmax-ui/src/app/modules/character/model/equipment/EnchantOptions.ts
import { Enchant } from "./Enchant";
import { EquippableItem } from "./EquippableItem";
import { Item } from "./Item";
import { ItemSubType } from "./ItemSubType";
import { ItemType } from "./ItemType";

export interface EnchantOptions {
	itemType: ItemType;
	itemSubType: ItemSubType;
	enchants: Enchant[];
}

export function getMatchingEnchantOptions(enchantOptions: EnchantOptions[], equippedItem: EquippableItem) {
	return enchantOptions.find(x => isMatchingEnchant(x, equippedItem.item))?.enchants || []
}

function isMatchingEnchant(x: EnchantOptions, item: Item) {
	return x.itemType === item.itemType && x.itemSubType === item.itemSubType
}

File: services/wow-minmax/src/main/java/wow/minmax/model/options/EnchantOptions.java
package wow.minmax.model.options;

import wow.commons.model.categorization.ItemSubType;
import wow.commons.model.categorization.ItemType;
import wow.commons.model.item.Enchant;

import java.util.List;

/**
 * User: POlszewski
 * Date: 2025-08-30
 */
public record EnchantOptions(
		ItemType itemType,
		ItemSubType itemSubType,
		List<Enchant> enchants
) {
}
